rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,633 @@
+                        from_asset=A_LUSD,
+                        to_asset=A_USD,
+                        timestamp=timestamp,
+                    )
+                    eth_price = PriceHistorian().query_historical_price(
+                        from_asset=A_ETH,
+                        to_asset=A_USD,
+                        timestamp=timestamp,
+                    )
+                    debt_after_amount = deserialize_optional_fval(change['debtAfter'], 'debtAfter', 'liquity')  # noqa: E501
+                    collateral_after_amount = deserialize_optional_fval(change['collateralAfter'], 'collateralAfter', 'liquity')  # noqa: E501
+                    event = LiquityTroveEvent(
+                        kind='trove',
+                        tx=change['transaction']['id'],
+                        address=owner,
+                        timestamp=timestamp,
+                        debt_after=AssetBalance(
+                            asset=A_LUSD,
+                            balance=Balance(
+                                amount=debt_after_amount,
+                                usd_value=lusd_price * debt_after_amount,
+                            ),
+                        ),
+                        collateral_after=AssetBalance(
+                            asset=A_ETH,
+                            balance=Balance(
+                                amount=collateral_after_amount,
+                                usd_value=eth_price * collateral_after_amount,
+                            ),
+                        ),
+                        debt_delta=AssetBalance(
+                            asset=A_LUSD,
+                            balance=Balance(
+                                amount=debt_change,
+                                usd_value=lusd_price * debt_change,
+                            ),
+                        ),
+                        collateral_delta=AssetBalance(
+                            asset=A_ETH,
+                            balance=Balance(
+                                amount=collateral_change,
+                                usd_value=eth_price * collateral_change,
+                            ),
+                        ),
+                        trove_operation=operation,
+                    )
+                    result[owner]['trove'].append(event)
+                except (DeserializationError, KeyError) as e:
+                    log.debug(f'Failed to deserialize Liquity trove event: {change}')
+                    msg = str(e)
+                    if isinstance(e, KeyError):
+                        msg = f'Missing key entry for {msg}.'
+                    self.msg_aggregator.add_warning(
+                        f'Ignoring Liquity Trove event in Liquity. '
+                        f'Failed to decode remote information. {msg}.',
+                    )
+                    continue
+
+        for stake in staked['lqtyStakes']:
```

Comment:
What if `lqtyStakes` is missing? You should log an error and fail nicely, no?

```suggestion
        for stake in staked['lqtyStakes']:
```